packages/botonic-react/src/webchat/hooks.js

```diff
@@ -71,11 +71,14 @@ export function useWebchat() {
     webchatReducer,
     webchatInitialState
   )
+  const [unreadMessages, setUnreadMessages] = useState(0)
 
   const addMessage = message =>
     webchatDispatch({ type: ADD_MESSAGE, payload: message })
-  const addMessageComponent = message =>
+  const addMessageComponent = message => {
+    setUnreadMessages(value => value + 1)
```

Comment:
Just as an observation, this is not 100% correct because not all messages will be unread messages and we're assuming this here.

I know this will work because everytime the webchat is closed this counter is set to 0, so no bubble will be displayed at first with the unread messages when the webchat is closed, but the counter will have a wrong value when the webchat is opened, am I right?

If we can check if the webchat is open/close at this point, maybe it's better to only increase the `unreadMessages` when the webchat is closed, what do you think?

Reply:
Yes you are right, I have refactored all this logic to have inside the webchatstate the value of unreadMessages and increment it taking into account the ack value.